internal/gengapic/generator.go

```diff
@@ -302,3 +304,44 @@ func (g *generator) nestedName(nested pbinfo.ProtoType) string {
 
 	return name
 }
+
+// autoPopulatedFields returns an array of snake-case MethodDescriptorProto
+// input field names that are specified for auto-population per the following
+// restrictions:
+//
+// * The field is a top-level string field of a unary method's request message.
+// * The field is not annotated with google.api.field_behavior = REQUIRED.
+// * The field name is listed in google.api.publishing.method_settings.auto_populated_fields.
+// * The field is annotated with google.api.field_info.format = UUID4.
+func (g *generator) autoPopulatedFields(servName string, m *descriptor.MethodDescriptorProto) []string {
+	var apfs []string
+	// Find the service config's AutoPopulatedFields entry by method name.
+	for _, s := range g.serviceConfig.GetPublishing().GetMethodSettings() {
+		if s.GetSelector() == g.fqn(m) {
+			apfs = s.AutoPopulatedFields
+			break
+		}
+	}
+	inType := g.descInfo.Type[*m.InputType].(*descriptor.DescriptorProto)
+	var validated []string
+	for _, apf := range apfs {
+		var field *descriptorpb.FieldDescriptorProto
+		// Find the input's FieldDescriptorProto by AutoPopulatedField name.
+		for _, f := range inType.GetField() {
+			if f.GetName() == apf {
+				field = f
+				break
+			}
+		}
+		// Do nothing and continue iterating unless all conditions above are met.
+		switch {
+		case field == nil:
+		case field.GetType() != fieldTypeString:
+		case field.GetLabel() == descriptor.FieldDescriptorProto_LABEL_REQUIRED:
```

Comment:
Taking the other thread here, if you need to check field behavior, you need to check the option, not the label - i know it's confusing.

So use the helper `isRequired` https://github.com/googleapis/gapic-generator-go/blob/2f1a8d3c867ec551e5a8bd1ce51e181ed4546319/internal/gengapic/helpers.go#L185-L200

`case isRequired(field):`

Reply:
Wow, yes, confusing. Will do, thanks!

Reply:
Fixed, thanks!